File: java/Arrays_Hashing/ProductExceptSelf.java
import java.util.*;

public class ProductExceptSelf{
    public static void main(String[] args){
        int[] nums = {1,2,4,6};

        int[] result = new int[nums.length];

        int prefix =1;
        for(int i = 0 ;i< nums.length; i++){
            result[i] = prefix;
            prefix *= nums[i];
        }
        int postfix = 1;
        for(int i = nums.length -1;i>=0;i--){
            result[i] *= postfix;
            postfix *= nums[i];
        }

        System.out.println(Arrays.toString(result));
    }
}

File: python/Arrays_Hashing/product_except_self.py
array = [1,2,4,6]

prefix = 1

result = []
for i in range(len(array)):
    result.append(prefix)
    prefix = prefix * array[i]

print(result)
postfix = 1

for i in range(len(array)-1,-1,-1):
    result[i] *= postfix
    postfix *= array[i]


print(result)